src/SprykerSdk/AsyncApi/Code/Builder/AsyncApiBuilder.php

```diff
@@ -551,4 +558,21 @@ protected function updateAsyncApi(string $targetFile, array $asyncApi): bool
 
         return $this->writeToFile($targetFile, $originAsyncApi);
     }
+
+    /**
+     * @param string $messageName
+     * @param array $messages
+     *
+     * @return bool
+     */
+    protected function messageNameExists($messageName, $messages): bool
+    {
+        foreach ($messages as $message) {
+            if ($message['$ref'] === sprintf('#/components/messages/%s', $messageName)) {
```

Comment:
You must use `sprintf('#/components/messages/%s', $messageName)` outside of the loop only once and not per iteration.